File: Blanquita-Backend/app/Models/BobinaPapel/VerProduccionBobinaTubo.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class VerProduccionBobinaTuboRequest(BaseModel):
    IdTipoBobina: Optional[int] = None


class VerProduccionBobinaTuboResponse(BaseModel):
    IdProduccionBobinaTubo: int
    IdTipoBobina: int
    NombreTipoBobina: str
    NombreEstadoProduccion: str
    CodigoBobina1: str
    CodigoBobina2: str
    NombreTurno: str
    FechaInicioProduccion: datetime
    CantidadLogsActual: int

File: Blanquita-Frontend/src/models/BobinaPapel/VerProduccionBobinaTubo.js
export class VerProduccionBobinaTuboRequest {
  constructor(IdTipoBobina = null) {
    this.IdTipoBobina = IdTipoBobina;
  }
}

export class VerProduccionBobinaTuboResponse {
  constructor(data) {
    this.IdProduccionBobinaTubo = data.IdProduccionBobinaTubo;
    this.IdTipoBobina = data.IdTipoBobina;
    this.NombreTipoBobina = data.NombreTipoBobina;
    this.NombreEstadoProduccion = data.NombreEstadoProduccion;
    this.CodigoBobina1 = data.CodigoBobina1;
    this.CodigoBobina2 = data.CodigoBobina2;
    this.NombreTurno = data.NombreTurno;
    this.FechaInicioProduccion = new Date(data.FechaInicioProduccion);
    this.CantidadLogsActual = data.CantidadLogsActual;
  }
}